src/BondingCurveRegistry.sol

```diff
@@ -38,52 +34,86 @@ contract BondingCurveRegistry is Initializable {
+    /// @param curveName The name of the curve
+    event BondingCurveAdded(uint256 indexed curveId, address indexed curveAddress, string indexed curveName);
+
+    /// @notice Emitted when the admin role is transferred
+    /// @param oldAdmin The previous admin address
+    /// @param newAdmin The new admin address
+    event OwnershipTransferred(address indexed oldAdmin, address indexed newAdmin);
 
     /* =================================================== */
-    /*                    INITIALIZER                      */
+    /*                    CONSTRUCTOR                      */
     /* =================================================== */
 
     /// @notice Initializes the BondingCurveRegistry contract
     /// @param _admin Address who may add curves to the registry
     /// NOTE: This function is called only once (during contract deployment)
-    function initialize(address _admin) external initializer {
-        require(_admin != address(0), "BondingCurveRegistry: requires owner");
+    constructor(address _admin) {
+        if (_admin == address(0)) {
+            revert Errors.BondingCurveRegistry_RequiresOwner();
+        }
         admin = _admin;
+        emit OwnershipTransferred(address(0), _admin);
     }
 
     /* =================================================== */
     /*               RESTRICTED FUNCTIONS                  */
     /* =================================================== */
 
-    /// @notice Set the admin address
-    /// @param _admin Address of the new admin
-    function setAdmin(address _admin) external onlyAdmin {
-        admin = _admin;
-    }
-
     /// @notice Add a new bonding curve to the registry
     /// @param bondingCurve Address of the new bonding curve
-    function addBondingCurve(address bondingCurve) external onlyAdmin {
+    function addBondingCurve(address bondingCurve) external {
+        if (msg.sender != admin) {
+            revert Errors.BondingCurveRegistry_OnlyOwner();
+        }
+
         if (curveIds[bondingCurve] != 0) {
             revert Errors.BondingCurveRegistry_CurveAlreadyExists();
         }
 
+        // Enforce curve name uniqueness
+        string memory curveName = IBaseCurve(bondingCurve).name();
+        if (registeredCurveNames[curveName]) {
+            revert Errors.BondingCurveRegistry_CurveNameNotUnique();
+        }
+
         // 0 is reserved to safeguard against uninitialized values
         ++count;
 
         // Add the curve to the registry, keeping track of its address and ID in separate tables
         curveAddresses[count] = bondingCurve;
         curveIds[bondingCurve] = count;
+
+        // Mark the curve name as registered
+        registeredCurveNames[curveName] = true;
+
+        emit BondingCurveAdded(count, bondingCurve, curveName);
+    }
+
+    /// @notice Transfer the admin role to a new address
+    /// @param newOwner The new admin address
+    function transferOwnership(address newOwner) external {
```

Comment:
3.13 - Shortcomings of Administrator Management in BondingCurveRegistry

It looks like we elected to allow ownership to be transferred to the zero address here, in order to make it clear if we decide that no further bonding curves may be added to a given instance of the registry.  I am okay with this.

@mihailo-maksa Please confirm if this was intentional, or if you think we should prevent this.